### ISSUE TITLE
TRTIS gRPU DRAM Out-Of-Memory

### ISSUE THREAD

**Issue by royinx** [NVIDIA/tensorrt-inference-server#563]
According to #207 ,  the case is similar but i have already using TensorRT model for TRTIS

I am trying to run the trtis on local pc 2070, with 2context with 1 tensorRT plan (yolo) 
using Python + async run + async get result
while testing the throughput of trtis, the memory raise rapidly. 

Even i shut down the client container, the DRAM stuck in 10GB out of 16GB 
- trtis off: 3.6GB
- trtis on: 6.8 GB
- after the 10requests: 16GB+2GB swap

if i keep sending the request and receive the result, the server DRAM still goes up and eat up all the swap. 

but VRAM keep the same with 6GB out of 8GB.

1). Is there any configuration i have to set inside config.pbtxt that release memory/cache?

2).another question, how can i know the trtis cpu usage and throughput of each model (since there are network latency on sending data)?

```
name: "yolov3_608_trt"
platform: "tensorrt_plan"
max_batch_size: 32
dynamic_batching {
  preferred_batch_size: [ 4, 8 ,16, 24, 32]
  max_queue_delay_microseconds: 100
}
input [
  {
    name: "000_net"
    data_type: TYPE_FP32
    format: FORMAT_NCHW
    dims: [ 3, 608, 608 ]
  }
]
output [
  {
    name: "082_convolutional"
    data_type: TYPE_FP32
    dims: [ 255, 19, 19 ]
  },
  {
    name: "094_convolutional"
    data_type: TYPE_FP32
    dims: [ 255, 38, 38 ]
  },
  {
    name: "106_convolutional"
    data_type: TYPE_FP32
    dims: [ 255, 76, 76 ]
  }
]


instance_group [
  {
    count:2
    kind: KIND_GPU
  }
]
```

**Comment by GuanLuo** [comment]
What version of TRTIS were you using and what network protocol did you use?

For 2), currently you can't know the per model CPU usage, but for throughput, you can use the [perf_client](https://github.com/NVIDIA/tensorrt-inference-server/blob/master/src/clients/c%2B%2B/perf_client.cc) which will report the request latency and other statistic for a given model.

**Comment by royinx** [comment]
That is container image - `nvcr.io/nvidia/tensorrtserver:19.06-py3`
TRTIS - **5.1.5**
protocol - **gRPC** + **8001**

2). thanks for the perf_client script, i will try it out. Actually I'm using python to write the pipeline, i will looking for the python API :D

**Comment by royinx** [comment]
Finally, I try the 3 version of `TRTIS 5.1.5: 19.05  19.06  19.07`
and run on both ONNX_runtime and TensorRT Runtime. (that means not from the model).

The same OOM issues comes up when using **gRPC** as protocol.

HTTP is fine 6.8GB -> 12GB -> 6.8GB

What should i do if using gRPC?

**Comment by GuanLuo** [comment]
#429 fixed this issue and it will be in 19.08 release. 19.08 should be released next week, and you can also build TRTIS from master branch if you want to test this early.

**Comment by deadeyegoodwin** [comment]
We also have more GRPC improvements in the pipeline.
You can also try using --grpc-infer-thread-count=16 and --grpc-stream-infer-thread-count=16 when launching the server.
